src/tx.rs

```diff
@@ -273,15 +273,8 @@ impl TransactionSkeleton {
         hash_inputs.append(&mut self.crossover.to_hash_inputs().to_vec());
 
         if let Some((cid, cdata)) = &self.call {
-            let mut state = Params::new().hash_length(64).to_state();
-
-            state.update(cid.as_bytes());
-            state.update(cdata);
-
-            let mut buf = [0u8; 64];
-            buf.copy_from_slice(state.finalize().as_ref());
-
-            hash_inputs.push(BlsScalar::from_bytes_wide(&buf));
+            hash_inputs.push(Hasher::digest(cid.as_bytes()));
+            hash_inputs.push(Hasher::digest(cdata));
         }
 
         hash_inputs
```

Comment:
I don't think you need all of this. You don't have to call the `poseidon_hash` for all of those scalar input, since it's very expensive. You should be able to use just use the `Hasher`'s `update` method. Only stuff that needs to be proven in the circuit would need it, so let's say that you need to prove the hash between the contract id and the transaction data, you can hash the transaction data with the `Hasher` and then just `hash` the contract id with the result. It would be way faster, and also the code will be reduced a lot.
If you don't need to prove the hash betweet the contract id and the data, then you can just use the `Hasher` for everything (but I think that is not the case, we should check the gadget).

Reply:
Looking at the [relevant part](https://github.com/dusk-network/rusk/blob/d62411299bc088cf9f8032ed6179f59eb7f8fda4/circuits/transfer/src/execute/variants.rs#L190-L204) of the gadget we can see that the hash of the transaction is constrained to be the same as the hash used to sign the note. In the interest of ensuring the TX remains non malleable (i.e. grant an attacker the possibility to swap call data), I suggest we keep this in here.